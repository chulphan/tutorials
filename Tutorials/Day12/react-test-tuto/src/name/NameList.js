import React, { Component } from 'react';


class NameList extends Component {

    static defaultProps = {
        namelist: []
    }

    render() {
        const { namelist } = this.props;
        const list = namelist.map((one, idx) => {
            return (<div key={idx}>
                {one.name}
            </div>);
        });
        return (
          <div>
              {list}
          </div>
        )
    }
}

export default NameList;