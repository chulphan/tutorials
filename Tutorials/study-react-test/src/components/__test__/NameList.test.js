import React from 'react';
import { shallow } from 'enzyme';
import '../../setupTests';

import NameList from '../NameList';

/*
    *********NameList 컴포넌트 테스트 항목***********
    NameList 컴포넌트가 에러없이 렌더링 되었는지
    가짜 데이터를 바탕으로 잘 렌더링 되었는지.
    renderList 메소드가 제대로 동작하는지.
    
*/

describe('about <NameList>..', () => {
    let component = null;
    it('NameList component render correctly.', () => {
        component = shallow(<NameList/>);
    });

    it('render correctly with mock data', () => {
        const mockData = ['testantil', 'testchul', 'testkim', 'testhwan'];

        component = shallow(<NameList names={mockData}/>);
        
        expect(component).toMatchSnapshot();

        expect(component.find('li')).toHaveLength(mockData.length);

        expect(component.find('li').contains(<li>testchul</li>)).toBe(true);

    });

    it('the renderList function is called correctly', () => {
        component = shallow(<NameList/>);
        component.setProps({names: ['1', '2', '3', '4', '5']});
        const renderList = component.instance().renderList();

        expect(renderList).toMatchSnapshot();
    });
})