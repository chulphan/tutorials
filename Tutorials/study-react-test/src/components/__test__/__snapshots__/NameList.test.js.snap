// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`about <NameList>.. render correctly with mock data 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      testantil
    </li>
    <li
      key="1"
    >
      testchul
    </li>
    <li
      key="2"
    >
      testkim
    </li>
    <li
      key="3"
    >
      testhwan
    </li>
  </ul>
</div>
`;

exports[`about <NameList>.. the renderList function is called correctly 1`] = `
Array [
  <li>
    1
</li>,
  <li>
    2
</li>,
  <li>
    3
</li>,
  <li>
    4
</li>,
  <li>
    5
</li>,
]
`;
